cadence/contracts/connectors/evm/UniswapV3SwapConnectors.cdc

```diff
@@ -0,0 +1,779 @@
+            if res == nil || res!.status != EVM.Status.successful { return nil }
+
+            let decoded = EVM.decodeABI(types: [Type<UInt256>()], data: res!.data)
+            if decoded.length == 0 { return nil }
+            let uintAmt = decoded[0] as! UInt256
+
+            let ercAddr = reverse
+                ? (out ? self.tokenPath[0] : self.tokenPath[self.tokenPath.length - 1])
+                : (out ? self.tokenPath[self.tokenPath.length - 1] : self.tokenPath[0])
+
+            return FlowEVMBridgeUtils.convertERC20AmountToCadenceAmount(uintAmt, erc20Address: ercAddr)
+        }
+
+        /// Executes exact input swap via router
+        access(self) fun _swapExactIn(exactVaultIn: @{FungibleToken.Vault}, amountOutMin: UFix64, reverse: Bool): @{FungibleToken.Vault} {
+            let id = self.uniqueID?.id?.toString() ?? "UNASSIGNED"
+            let idType = self.uniqueID?.getType()?.identifier ?? "UNASSIGNED"
+            let coa = self.borrowCOA()
+                ?? panic("Invalid COA Capability in V3 Swapper \(self.getType().identifier) ID \(idType)#\(id)")
+
+            // Bridge fee
+            let bridgeFeeBalance = EVM.Balance(attoflow: 0)
+            bridgeFeeBalance.setFLOW(flow: 2.0 * FlowEVMBridgeUtils.calculateBridgeFee(bytes: 256))
+            let feeVault <- coa.withdraw(balance: bridgeFeeBalance)
+            let feeVaultRef = &feeVault as auth(FungibleToken.Withdraw) &{FungibleToken.Vault}
+
+            // I/O tokens
+            let inToken = reverse ? self.tokenPath[self.tokenPath.length - 1] : self.tokenPath[0]
+            let outToken = reverse ? self.tokenPath[0] : self.tokenPath[self.tokenPath.length - 1]
+
+            // Bridge input to EVM
+            let evmAmountIn = FlowEVMBridgeUtils.convertCadenceAmountToERC20Amount(exactVaultIn.balance, erc20Address: inToken)
+            coa.depositTokens(vault: <-exactVaultIn, feeProvider: feeVaultRef)
+
+            // Build path
+            let pathBytes = self._buildPathBytes(reverse: reverse)
+
+            // Approve
+            var res = self._call(
+                to: inToken,
+                signature: "approve(address,uint256)",
+                args: [self.routerAddress, evmAmountIn],
+                gasLimit: 120_000,
+                value: 0
+            )!
+            if res.status != EVM.Status.successful {
+                UniswapV3SwapConnectors._callError("approve(address,uint256)", res, inToken, idType, id, self.getType())
+            }
+
+            // Slippage/min out on EVM units (adjust factor to your policy)
+            let slippage = 0.001 // 0.1%
+            let minOutUint = FlowEVMBridgeUtils.convertCadenceAmountToERC20Amount(
+                amountOutMin * (1.0 - slippage),
+                erc20Address: outToken
+            )
+
+            // exactInput((bytes,address,uint256,uint256)) selector = 0xb858183f
+            let selector: [UInt8] = [0xb8, 0x58, 0x18, 0x3f]
```

Comment:
It's not apparent that the selector values match the commented selector which on its own need to match the .sol contract selector. Still, I think it'd be easier to catch an issue with:


```suggestion
            // exactInput((bytes,address,uint256,uint256)) selector = 0xb858183f
            let selector: [UInt8] = "b858183f".decodeHex()
```